processing/src/main/java/io/druid/query/aggregation/LongMaxAggregatorFactory.java

```diff
@@ -72,18 +72,24 @@ public LongMaxAggregatorFactory(String name, String fieldName)
   }
 
   @Override
-  public Aggregator factorize(ColumnSelectorFactory metricFactory)
+  protected ColumnValueSelector selector(ColumnSelectorFactory metricFactory)
   {
-    return new LongMaxAggregator(getLongColumnSelector(metricFactory));
+    return getLongColumnSelector(metricFactory);
   }
 
   @Override
-  public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  protected Aggregator factorize(ColumnSelectorFactory metricFactory, ColumnValueSelector selector)
   {
-    return new LongMaxBufferAggregator(getLongColumnSelector(metricFactory));
+    return new LongMaxAggregator(selector);
   }
 
-  private BaseLongColumnValueSelector getLongColumnSelector(ColumnSelectorFactory metricFactory)
+  @Override
+  protected BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory, ColumnValueSelector selector)
+  {
+    return new LongMaxBufferAggregator(selector);
+  }
+
+  private ColumnValueSelector getLongColumnSelector(ColumnSelectorFactory metricFactory)
```

Comment:
Now this private method is used just in one other method. It could be inlined.

Reply:
done.